sys/net/link_layer/nomac/nomac.c

```diff
@@ -0,0 +1,295 @@
+
+        if (msg_cmd.type == NETDEV_MSG_EVENT_TYPE) {
+            dev->driver->event(dev, msg_cmd.content.value);
+        }
+        else if (msg_cmd.type == NETAPI_MSG_TYPE) {
+            cmd = (netapi_cmd_t *)(msg_cmd.content.ptr);
+            ack = cmd->ack;
+            msg_ack.content.ptr = (char *)ack;
+
+            switch (cmd->type) {
+                case NETAPI_CMD_SND:
+                    ack->result = _nomac_send(dev, (netapi_snd_pkt_t *)cmd);
+                    break;
+
+                case NETAPI_CMD_GET:
+                    ack->result = _nomac_get_option(dev, (netapi_conf_t *)cmd);
+                    break;
+
+                case NETAPI_CMD_SET:
+                    ack->result = _nomac_set_option(dev, (netapi_conf_t *)cmd);
+                    break;
+
+                case NETAPI_CMD_REG:
+                    ack->result = -ENOBUFS;
+
+                    for (int i = 0; i < NOMAC_REGISTRY_SIZE; i++) {
+                        if (_nomac_registry[i].registrar_pid == KERNEL_PID_UNDEF) {
+                            netapi_reg_t *reg = (netapi_reg_t *)cmd;
+
+                            _nomac_registry[i].registrar_pid = thread_getpid();
+                            _nomac_registry[i].recipient_pid = reg->reg_pid;
+                            ack->result = NETAPI_STATUS_OK;
+
+                            break;
+                        }
+                    }
+
+                    break;
+
+                case NETAPI_CMD_UNREG:
+                    ack->result = NETAPI_STATUS_OK;
+
+                    for (int i = 0; i < NOMAC_REGISTRY_SIZE; i++) {
+                        netapi_reg_t *reg = (netapi_reg_t *)cmd;
+
+                        if (_nomac_registry[i].registrar_pid == thread_getpid() &&
+                            _nomac_registry[i].recipient_pid == reg->reg_pid) {
+                            _nomac_registry[i].recipient_pid = KERNEL_PID_UNDEF;
+
+                            break;
+                        }
+
+                    }
+
+                    break;
+
+#ifdef MODULE_NETDEV_DUMMY
+
+                case NETAPI_CMD_FIRE_RCV:
+                    ack->result = unittest_netdev_dummy_fire_rcv_event(dev,
+                                  ((netapi_rcv_pkt_t *)cmd)->src,
+                                  ((netapi_rcv_pkt_t *)cmd)->src_len,
+                                  ((netapi_rcv_pkt_t *)cmd)->dest,
+                                  ((netapi_rcv_pkt_t *)cmd)->dest_len,
+                                  ((netapi_rcv_pkt_t *)cmd)->data,
+                                  ((netapi_rcv_pkt_t *)cmd)->data_len);
+                    break;
+#endif
+
+                default:
+                    ack->result = -ENOTSUP;
+                    break;
+            }
+
+            ack->type = NETAPI_CMD_ACK;
+            ack->orig = cmd->type;
+            msg_reply(&msg_cmd, &msg_ack);
+        }
+    }
+
+    /* never reached */
+    return NULL;
+}
+
+void nomac_init_module(void)
```

Comment:
Same as above. If there is only one nomac_registry for multiple nomac threads, moving this initialization into the thread is no option.